test/utils.py

```diff
@@ -21,6 +21,22 @@ def needs_cuda(test_item):
     return test_item
 
 
+def cpu_and_cuda():
+    # TODO: Figure out how to dedupe this logic in needs_cuda() and here.
```

Comment:
I think we should make `cpu_and_cuda()` and `needs_cuda()` be exactly like they are in torchvision, and just handle the skip and `FAIL_WITHOUT_CUDA` within https://github.com/pytorch/vision/blob/80edd48291aa83eb57d90a396e57e036f6bb7fe8/test/conftest.py#L25C5-L25C34. Our version of that would be a lot simpler than that of torchvision though, don't be scared.


I can take care of it if you want (but I'd prefer doing this change within this PR)